src/workerd/io/worker.c++

```diff
@@ -631,29 +648,31 @@ struct Worker::Isolate::Impl {
         i.get()->contextCreated(v8_inspector::V8ContextInfo(context, 1, toStringView("Worker")));
       }
 
-      if (impl.inspector == kj::none) {
-        // When not running in preview mode, we replace the default V8 console.log(), etc. methods,
-        // to give the worker access to logged content.
-        auto global = context->Global();
-        auto consoleStr = jsg::v8StrIntern(lock->v8Isolate, "console");
-        auto console = jsg::check(global->Get(context, consoleStr));
-
-        auto setHandler = [&](const char* method, LogLevel level) {
-          auto methodStr = jsg::v8StrIntern(lock->v8Isolate, method);
-
-          auto f = lock->wrapSimpleFunction(context,
-              [level](jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>& info) {
-            handleLog(js, level, info);
-          });
-          jsg::check(console.As<v8::Object>()->Set(context, methodStr, f));
-        };
+      // We replace the default V8 console.log(), etc. methods, to give the worker access to
+      // logged content, and log formatted values to stdout/stderr locally.
+      auto global = context->Global();
+      auto consoleStr = jsg::v8StrIntern(lock->v8Isolate, "console");
+      auto console = jsg::check(global->Get(context, consoleStr)).As<v8::Object>();
+      auto mode = consoleMode;
 
-        setHandler("debug", LogLevel::DEBUG_);
-        setHandler("error", LogLevel::ERROR);
-        setHandler("info", LogLevel::INFO);
-        setHandler("log", LogLevel::LOG);
-        setHandler("warn", LogLevel::WARN);
-      }
+      auto setHandler = [&](const char* method, LogLevel level) {
+        auto methodStr = jsg::v8StrIntern(lock->v8Isolate, method);
+        v8::Global<v8::Function> original(
+          lock->v8Isolate, jsg::check(console->Get(context, methodStr)).As<v8::Function>());
+
+        auto f = lock->wrapSimpleFunction(context,
+            [mode, level, original = kj::mv(original)](jsg::Lock& js,
+                const v8::FunctionCallbackInfo<v8::Value>& info) {
+          handleLog(js, mode, level, original, info);
+        });
+        jsg::check(console->Set(context, methodStr, f));
+      };
+
+      setHandler("debug", LogLevel::DEBUG_);
+      setHandler("error", LogLevel::ERROR);
+      setHandler("info", LogLevel::INFO);
+      setHandler("log", LogLevel::LOG);
+      setHandler("warn", LogLevel::WARN);
```

Comment:
Good question. This particular bit predates me so I'm not sure and I haven't looked into it yet. @kentonv might have some insight. It might be that `ConsoleDelegate` would just work but no one has tried? Really not sure.